src/sync.js

```diff
@@ -237,8 +237,8 @@ class Sync {
       if (err) { log.error(err) }
       if (err && err.code === 'ENOSPC') {
         callback(new Error('The disk space on your computer is full!'))
-      } else if (err && err.message === REVOKED) {
-        callback(err)
+      } else if (err && err.status === 400) {
+        callback(new Error(REVOKED))
```

Comment:
Cozy-desktop won't try to move one directory inside one of its sub-directories. But, yes, it's not ideal to detect a revocation with a 400 status code. It's most reliable way to detect it for the moment, but not totally safe. I keep in mind to improve it in cozy-stack.